CMakeLists.txt

```diff
@@ -472,7 +472,7 @@ if (NOT MSVC)
     endif()
 endif()
 
-if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64" OR ${CMAKE_GENERATOR_PLATFORM_LWR} MATCHES "arm64")
+if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64" OR "${CMAKE_GENERATOR_PLATFORM_LWR}" MATCHES "arm64")
```

Comment:
why? CMAKE_SYSTEM_PROCESSOR isn't quoted either.

Reply:
The build was failing on e.g. Linux amd64 because CMAKE_GENERATOR_PLATFORM_LWR was an empty string, and unquoted empty strings are the same as the variable not being there - causing a syntax error.